lua/nvim-possession/init.lua

```diff
@@ -125,7 +126,32 @@ M.setup = function(user_opts)
 			return
 		end
 
-		return fzf.files({
+		local function list_sessions(fzf_cb)
+			local sessions = {}
+			for name, type in vim.fs.dir(user_config.sessions.sessions_path, { depth = math.huge }) do
+				if type == "file" then
+					local stat = vim.uv.fs_stat(user_config.sessions.sessions_path .. name)
+					if stat then
+						table.insert(sessions, { name = name, mtime = stat.mtime })
+					end
+				end
+			end
+			table.sort(sessions, function(a, b)
+				if a.mtime.sec ~= b.mtime.sec then
+					return a.mtime.sec > b.mtime.sec
+				end
+				if a.mtime.nsec ~= b.mtime.nsec then
+					return a.mtime.nsec > b.mtime.nsec
+				end
+				return a.name < b.name
+			end)
+			for _, sess in ipairs(sessions) do
+				fzf_cb(sess.name)
```

Comment:
Out of curiosity could you explain this constructor? I understand you are now re-assigning the session names to a new table via a lambda construct `fzf_cb` which is defined as function argument at line 129?

Reply:
Yes, `fzf_cb` is passed to `list_session` by `fzf_exec`.